### PR TITLE
Ensure deployment_id is set to provided deployment, required for Azure OpenAI.

### LINKED ISSUE #5001: Azure OpenAI Embeddings failed due to no deployment_id set.
### System Info

Broken by #4915 

Error: `Must provide an 'engine' or 'deployment_id' parameter to create a <class 'openai.api_resources.embedding.Embedding'>`

I'm putting a PR out to fix this now.

### Who can help?

_No response_

### Information

- [X] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [X] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Run example notebook: https://github.com/hwchase17/langchain/blob/22d844dc0795e7e53a4cc499bf4974cb83df490d/docs/modules/models/text_embedding/examples/azureopenai.ipynb

### Expected behavior

Embedding using Azure OpenAI should work.

### PR DESCRIPTION
**Pull request by lupickup**
# Ensure deployment_id is set to provided deployment, required for Azure OpenAI.

Fixes #5001 

## Before submitting

Verified in my own code and by running this example notebook, works after the fix (as in it doesn't fail before trying to talk to the endpoint): https://github.com/hwchase17/langchain/blob/22d844dc0795e7e53a4cc499bf4974cb83df490d/docs/modules/models/text_embedding/examples/azureopenai.ipynb

## Who can review?

@agola11 @hwchase17

### CODE REVIEW

**Comment by dev2049** [comment]
should we just revert #4915? seems like the api update they were referring to was specific to completion endpoints, so we shouldn't have changed in first place

**Comment by tot0** [comment]
Also valid, I’m out for most of the evening so won’t be able to update this
PR till after 10pm (ideally could keep the test?)

Revert also works though so if fix can’t wait revert seems easiest.

Cheers,
Lucas

**Comment by dev2049** [comment]
will revert for now and we can add tests in follow-up

**Comment by tot0** [comment]
@dev2049 Updated this PR to only have unit test.

File: tests/unit_tests/models/test_openai_embeddings.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    os.environ["OPENAI_API_BASE"] = "https://your-endpoint.openai.azure.com/"
+    os.environ["OPENAI_API_KEY"] = "your AzureOpenAI key"
+    os.environ["OPENAI_API_VERSION"] = "2023-03-15-preview"
+
+    embeddings = OpenAIEmbeddings(deployment="your-embeddings-deployment-name")
+    text = "This is a test document."
+
+    try:
+        query_result = embeddings.embed_query(text)
+    except error.InvalidRequestError as e:
```
Comment (hwchase17) [inline tests/unit_tests/models/test_openai_embeddings.py:19]: sorry what is this testing?

Comment (tot0) [reply tests/unit_tests/models/test_openai_embeddings.py:19]: That deployment_id/engine is set on openai.Embeddings when the api_type is `azure`. Basically a regression test against a similar change being made as the one that was reverted in response to this PR.

**Pull request merged**